tests/filecheck/transforms/eqsat-create-eclasses.mlir

```diff
@@ -0,0 +1,9 @@
+// RUN: xdsl-opt -p eqsat-create-eclasses --verify-diagnostics %s | filecheck %s
+
+func.func @test(%x : index) -> (index) {
+    %c2 = arith.constant 2 : index
+    %res = arith.muli %x, %c2 : index
+    func.return %res : index
+}
+
+// CHECK: Error while applying pattern: Ops with non-single results not handled
```

Comment:
Oh interesting, I forgot about function returns, they seem useful enough to have from the get-go, could you please change the check to rather be `len(results) > 1`? I would then recommend using the `test` dialect in filecheck, to make sure we're testing exactly what we mean to test:
```suggestion
func.func @test(%x : index) -> (index) {
    %r0, %r1 = "test.op"() : () -> (index, index)
}

// CHECK: Error while applying pattern: Only single-result operations can be converted to eclasses, got 2 results
```
(I also changed the message do include the length of the results, should hopefully help debug this in the future)